hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/functional/TestHoodieSparkMergeOnReadTableRollback.java

```diff
@@ -516,8 +556,6 @@ void testMORTableRestore(boolean restoreAfterCompaction) throws Exception {
     JavaRDD<HoodieRecord> writeRecords = jsc().parallelize(records, 1);
     JavaRDD<WriteStatus> writeStatusJavaRDD = client.upsert(writeRecords, newCommitTime);
     client.commit(newCommitTime, writeStatusJavaRDD);
-    List<WriteStatus> statuses = writeStatusJavaRDD.collect();
-    assertNoWriteErrors(statuses);
```

Comment:
Here I think you can do similar checks like other tests you modified:
```
      statuses = writeStatusJavaRDD.collect();
      // Verify there are no errors
      assertNoWriteErrors(statuses);

      client.commit(newCommitTime, jsc().parallelize(statuses));
```

Reply:
Yeah, this is a coarse hack though (just to keep the tests mostly intact). We should not actually dereference RDDs here since this couples tests not with API but with current implementation that might be subject to change in the future